### PR TITLE
2508: Logging Exporter Example for Trace and Metrics

### PR DESCRIPTION
**Pull request by reggiemcdonald**
This PR adds an example to the examples directory showing how to make use of the logging exporters for trace and metrics in a single runnable example. I thought this would help to address #2508 

The jaeger example was used as a basis for this example, altered to make use of the logging exporters

### CODE REVIEW

File: examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   *
+   * @return A ready-to-use {@link OpenTelemetry} instance.
+   */
+  public static OpenTelemetry initOpenTelemetry() {
+    // This will be used to create instruments
+    SdkMeterProvider meterProvider = SdkMeterProvider.builder().buildAndRegisterGlobal();
+
+    // Create an instance of IntervalMetricReader and configure it
+    // to read metrics from the meterProvider and export them to the logging exporter
+    IntervalMetricReader.builder()
```
Comment (bogdandrutu) [inline examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java:33]: Confused what you do with the result of the `build()`?

Comment (reggiemcdonald) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java:33]: I don't actually use the result. However, calling `build()` creates an instance of `IntervalMetricReader` and thus starts it: 
https://github.com/open-telemetry/opentelemetry-java/blob/e363e891dca81a8b07045cfe524990b639882d39/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java#L61-L70

Comment (bogdandrutu) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java:33]: facepalm :)

File: examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        .setExportIntervalMillis(METRIC_EXPORT_INTERVAL_MS)
+        .build();
+
+    // Tracer provider configured to export spans with SimpleSpanProcessor using
+    // the logging exporter.
+    SdkTracerProvider tracerProvider =
+        SdkTracerProvider.builder()
+            .addSpanProcessor(SimpleSpanProcessor.create(new LoggingSpanExporter()))
+            .build();
+    return OpenTelemetrySdk.builder().setTracerProvider(tracerProvider).build();
```
Comment (bogdandrutu) [inline examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java:45]: Register to global to be consistent with metrics?

**Review by bogdandrutu — APPROVED** [review]

File: examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java (hunk 2 above)
Comment (anuraaga) [inline examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java:45]:
```suggestion
    return OpenTelemetrySdk.builder().setTracerProvider(tracerProvider).buildAndRegisterGlobal();
```

**Review by anuraaga — APPROVED** [review]
Thanks for the help @reggiemcdonald!

File: examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java (hunk 2 above)
Comment (reggiemcdonald) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java:45]: Yep good call

File: examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java [hunk 3]
```diff
@@ -16,8 +16,8 @@
   private final Tracer tracer;
   private final LongCounter counter;
 
-  public LoggingExporterExample(OpenTelemetry openTelemetry) {
-    tracer = openTelemetry.getTracer(INSTRUMENTATION_NAME);
+  public LoggingExporterExample() {
+    tracer = GlobalOpenTelemetry.getTracer(INSTRUMENTATION_NAME);
```
Comment (anuraaga) [inline examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:20]: Ah sorry for the confusion, we still use the returned instance here. The global is only to support special instrumentation such as mysql.

Comment (reggiemcdonald) [reply examples/logging/src/main/java/io/opentelemetry/example/logging/LoggingExporterExample.java:20]: gotcha sorry will revert

File: examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    // This will be used to create instruments
+    SdkMeterProvider meterProvider = SdkMeterProvider.builder().buildAndRegisterGlobal();
+
+    // Create an instance of IntervalMetricReader and configure it
+    // to read metrics from the meterProvider and export them to the logging exporter
+    IntervalMetricReader.builder()
+        .setMetricExporter(new LoggingMetricExporter())
+        .setMetricProducers(Collections.singleton(meterProvider))
+        .setExportIntervalMillis(METRIC_EXPORT_INTERVAL_MS)
+        .build();
```
Comment (jkwatson) [inline examples/logging/src/main/java/io/opentelemetry/example/logging/ExampleConfiguration.java:37]: It does make me think that we should rename this to `start()` to make it super clear that it starts up the background thread.

**Review by jkwatson — APPROVED** [review]

**Pull request merged**